### PR TITLE
Flora (like trees, rocks and etc.) now drops materials when destroyed.

### PR DESCRIPTION
**Pull request by Helg2**
## About The Pull Request
I find it strange that you don't get anything if you just break tree or rock with brute force. So you will now get something but with 0.6 multiplier.
I chose 0.6 multiplier so using tools are still preferable, but you still get amount of recourses you can do something with.

## Why It's Good For The Game
You can gather some materials if you have no tools besides the toolbox.

## Changelog

:cl:
qol: Trees, rocks, grass and etc. now drop materials when destroyed.
/:cl:

### CODE REVIEW

**Comment by Farquaar** [comment]
This is definitely really cool, but I think it's more of a consistency fix than a qol change.

**Comment by Helg2** [comment]
Well, dunno

File: code/game/objects/structures/flora.dm [hunk 1]
```diff
@@ -263,6 +268,12 @@
 	var/matrix/M = matrix(transform)
 	transform = M.Turn(-previous_rotation)
 
+/obj/structure/flora/deconstruct()
+	if(!(flags_1 & NODECONSTRUCT_1))
+		harvest()
+		destroyed = TRUE
```
Comment (Paxilmaniac) [inline code/game/objects/structures/flora.dm:274]: If destroyed is only set to true after harvest is called, and the only place destroyed is checked is inside harvest, then wouldn't that mean destroying the flora would always drop the full amount of the results?

**Comment by Helg2** [comment]
Seems to work fine

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/flora.dm [hunk 2]
```diff
@@ -263,6 +268,15 @@
 	var/matrix/M = matrix(transform)
 	transform = M.Turn(-previous_rotation)
 
+/obj/structure/flora/deconstruct()
+	if(!(flags_1 & NODECONSTRUCT_1))
+		if(harvested)
+			return ..()
+
+		destroyed = TRUE
+		harvest()
```
[the remark this answers is not in the document]
Comment (Helg2) [reply code/game/objects/structures/flora.dm]: Okie

**Comment by optimumtact** [comment]
what would a grass drop lol

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/flora.dm [hunk 3]
```diff
@@ -263,6 +265,15 @@
 	var/matrix/M = matrix(transform)
 	transform = M.Turn(-previous_rotation)
 
+/obj/structure/flora/deconstruct()
+	if(!(flags_1 & NODECONSTRUCT_1))
+		if(harvested)
+			return ..()
+
+		product_amount_multiplier = 0.6
+		harvest()
```
[the remark this answers is not in the document]
Comment (Helg2) [reply code/game/objects/structures/flora.dm:274]: Oh

**Comment by Pumpkinoe** [comment]
Does this work fine with mining rocks on icebox for sand?

**Comment by Helg2** [comment]
> Does this work fine with mining rocks on icebox for sand?

Yes

File: code/game/objects/structures/flora.dm [hunk 4]
```diff
@@ -193,7 +193,7 @@
 	//If it *is* an item stack, we don't want to go through 50 different iterations of a new object where it just gets qdeleted after the first
 	. = list()
 	for(var/product in products_to_create)
-		var/amount_to_create = products_to_create[product]
+		var/amount_to_create = products_to_create[product]*product_amount_multiplier
```
Comment (Fikou) [inline code/game/objects/structures/flora.dm:196]: round this?

Comment (Helg2) [reply code/game/objects/structures/flora.dm:196]: ?

Comment (JohnFulpWillard) [reply code/game/objects/structures/flora.dm:196]:
```suggestion
		var/amount_to_create = round(products_to_create[product]*product_amount_multiplier, 1)
```

So you're not creating 1.2 products or something

Comment (Helg2) [reply code/game/objects/structures/flora.dm]: okay

**Pull request merged**